subsys/shell/modules/device_service.c

```diff
@@ -79,7 +79,17 @@ static int cmd_device_list(const struct shell *sh,
 #endif /* CONFIG_PM_DEVICE */
 		}
 
+#ifdef CONFIG_PM_DEVICE_RUNTIME
+		if (dev->pm != NULL) {
+			uint32_t usage = pm_device_usage_get(dev);
```

Comment:
you should avoid accessing dev->pm, just provide a dummy impl for pm_device_usage_get in case PM is not built-in. Function could return int, <0 for error (-ENOSYS), >=0 for usage counter. It will also avoid ifdeffery in clients.

Reply:
wow, what was I thinking... there were other issues as well, namely returning a counter for device with normal pm, non-runtime, what's the opposite of "runtime" in this context? whatever, fixed, should handle all cases now, example:

```console
uart:~$ device list 
devices:
- clock@40000000 (READY)
- gpio@50000000 (READY)
- random@4000d000 (READY)
- entropy_bt_hci (READY)
- uart@40002000 (active)
- flash-controller@4001e000 (READY)
- i2c@40003000 (active)
- buttons (suspended, usage=0)
- leds (READY)
```